File: output/TestRefactor/scripts/menu.py
#!/usr/bin/env python3
"""
Project menu for TESTREFACTOR
Usage: python scripts/menu.py
"""
import subprocess
import sys
from pathlib import Path

ROOT = Path(__file__).parent.parent

COMMANDS = [
    ("1", "install       → pip install -r requirements.txt", "pip install -r requirements.txt"),
    ("2", "run           → python src/main.py", "python src/main.py"),
    ("3", "test          → pytest", "pytest"),
    ("4", "coverage      → pytest --cov=src --cov-report=term", "pytest --cov=src --cov-report=term"),
    ("5", "format        → black src/ tests/", "black src/ tests/"),
    ("6", "format:lic    → python scripts/format_with_licenses.py", "python scripts/format_with_licenses.py"),

    ("8", "pre-commit    → format + format:lic", "black src/ tests/ && python scripts/format_with_licenses.py"),
    ("0", "exit", None),
]

def run(cmd):
    print(f"\n> {cmd}\n")
    try:
        subprocess.run(cmd, shell=True, check=True, cwd=ROOT)
    except subprocess.CalledProcessError as e:
        print(f"Command failed with exit code {e.returncode}")

def show_menu():
    print("\n========================================")
    print(f"  TESTREFACTOR — Project Menu")
    print("========================================")
    for key, label, _ in COMMANDS:
        if key != "0":
            print(f"  [{key}] {label}")
    print("  [0] exit")
    print("----------------------------------------")

def main():
    while True:
        show_menu()
        choice = input("  Select: ").strip()
        match = next((c for c in COMMANDS if c[0] == choice), None)
        if not match:
            print("  Invalid choice.")
        elif match[2] is None:
            print("  Goodbye!")
            sys.exit(0)
        else:
            run(match[2])

if __name__ == "__main__":
    main()

File: output/Test_Tasks_MD/scripts/menu.js
#!/usr/bin/env node
/**
 * Project menu for test_tasks_md
 * Usage: node scripts/menu.js
 */
const { execSync } = require('child_process');
const readline = require('readline');

const COMMANDS = [
  { key: "1", label: "install       → npm install", cmd: "npm install" },
  { key: "2", label: "start         → node src/js/index.js", cmd: "node src/js/index.js" },
  { key: "3", label: "test          → npm test", cmd: "npm test" },
  { key: "4", label: "coverage      → jest --coverage", cmd: "jest --coverage" },
  { key: "5", label: "format        → prettier --write "src/**/*.js"", cmd: `prettier --write "src/**/*.js"` },
  { key: "6", label: "format:lic    → node scripts/format-with-licenses.js", cmd: "node scripts/format-with-licenses.js" },

  { key: "8", label: "pre-commit    → format + format:lic", cmd: "npm run format && node scripts/format-with-licenses.js" },

  { key: "0", label: "exit", cmd: null },
];

function run(cmd) {
  console.log(`\n> ${cmd}\n`);
  try {
    execSync(cmd, { stdio: 'inherit', cwd: require('path').join(__dirname, '..') });
  } catch (e) {
    console.error('Command failed with exit code', e.status);
  }
}

function showMenu() {
  console.log('\n========================================');
  console.log('  test_tasks_md — Project Menu');
  console.log('========================================');
  COMMANDS.forEach(c => c.key !== "0" && console.log(`  [${c.key}] ${c.label}`));
  console.log('  [0] exit');
  console.log('----------------------------------------');
}

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

function prompt() {
  showMenu();
  rl.question('  Select: ', (answer) => {
    const entry = COMMANDS.find(c => c.key === answer.trim());
    if (!entry) {
      console.log('  Invalid choice.');
    } else if (entry.cmd === null) {
      console.log('  Goodbye!');
      rl.close();
      return;
    } else {
      run(entry.cmd);
    }
    prompt();
  });
}

prompt();
